src/main/java/org/zalando/nakadi/service/subscription/SubscriptionStreamerFactory.java

```diff
@@ -59,16 +56,15 @@ public SubscriptionStreamerFactory(
     }
 
     public SubscriptionStreamer build(
-            final String subscriptionId,
+            final Subscription subscription,
             final StreamParameters streamParameters,
             final SubscriptionOutput output,
             final AtomicBoolean connectionReady,
-            final BlacklistService blacklistService) throws NoSuchSubscriptionException, ServiceUnavailableException,
+            final BlacklistService blacklistService,
+            final Client client) throws NoSuchSubscriptionException, ServiceUnavailableException,
             InternalNakadiException, NoSuchEventTypeException {
-
-        final Subscription subscription = subscriptionDbRepository.getSubscription(subscriptionId);
         final Session session = Session.generate(1);
-        final String loggingPath = "subscription." + subscriptionId + "." + session.getId();
+        final String loggingPath = "subscription." + subscription.getId() + "." + session.getId();
```

Comment:
@v-stepanov to move the authorization outside of the builder, which is a very weird layer to make authorization.